pkg/collector/corechecks/cluster/ksm/kubernetes_state_transformers.go

```diff
@@ -349,6 +349,23 @@ func jobStatusSucceededTransformer(s aggregator.Sender, name string, metric ksms
 
 // jobStatusFailedTransformer sends a metric based on kube_job_status_failed
 func jobStatusFailedTransformer(s aggregator.Sender, name string, metric ksmstore.DDMetric, hostname string, tags []string) {
+	// Remove the `reason` tag to reduce the cardinality
+	reasonTagIndex := -1
+	for idx, tag := range tags {
+		if strings.HasPrefix(tag, "reason:") {
+			reasonTagIndex = idx
+			break
+		}
+	}
+
+	if reasonTagIndex != -1 && metric.Val == 0 {
```

Comment:
These (no reason tag and value 0) were reported before and now they're not. Why? (Maybe this is obvious, but it's my first time looking at this code).

Also, I think it might be worth to add a test for the general case and this one given that we already have tests for this function in `Test_jobStatusFailedTransformer`.

Reply:
In fact, there was no `reason` tag on the metric with the legacy python ksm check.
The `reason` tag was introduced in KSM v2.

Reply:
@davidor I’ve added a comment explaining the behaviour of the upstream metric to better understand the logic and I’ve updated the test according to this change. 👍🏻